src/api/controllers/user-controllers.ts

```diff
@@ -79,6 +92,53 @@ const updateUserProfile = async (req: Request, res: Response, next: NextFunction
     }
 };
 
+const addFavoriteActivity = async (req: Request, res: Response, next: NextFunction) => {
+    /** Get the user firebase id (identifier) and the id of the activity we want to favorite (update) */
+    const { identifier } = req.body;
+    var { update } = req.body;
+
+    /** Verify user exists */
+    const userCheck = await userServices.getUserById(identifier.firebase_id);
+
+    /** If user exists */
+    if (userCheck.found && userCheck.user) {
+        /** array to be sent to the update method in the service layer */
+        let favoriteActivitiesOld = userCheck.user.favoriteActivities;
+
+        /** If the user already has a list of favorite activities, then we will modify the current list and send it to the service layer*/
+        if (userCheck.user.favoriteActivities) {
+            const index = favoriteActivitiesOld.findIndex((element) => element == update.favoriteActivities);
+
+            /** If the current activity was not already favorited, then add it to the user's list of favorites*/
+            if (index == -1) {
+                favoriteActivitiesOld.push(update.favoriteActivities);
+            } /** If the current activity was already favorited in the past, then simply un-favorite it*/ else {
+                favoriteActivitiesOld.splice(index, 1);
+            }
+        } /** If the user had not favorited any activity before this, then we create a new array and add our current activity on it, then send this array to the service layer to update the user model */ else {
+            favoriteActivitiesOld = [update.favoriteActivities];
+        }
+
+        /** Assigning our new array to the update variable so it can be sent to the service layer and update the user's list of favorite activities**/
+        update = { favoriteActivities: favoriteActivitiesOld };
+
+        console.log(update);
+
+        try {
```

Comment:
put `try {} catch (e) {next(e)}` clause at the outmost, so that if there's any error the `NextFunction` could catch it

Reply:
Done